managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -100,8 +100,9 @@ public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
         } else if (cursor.config.isAutoSkipNonRecoverableData() && exception instanceof NonRecoverableLedgerException) {
```

Comment:
I think we should only apply this patch only if the BK error code is `Code.NoSuchEntryException`. See

https://github.com/AnonHxy/pulsar/blob/36ee4d757f9511788519228e573eff91f7c7e697/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L3747

Reply:
OK, I will fix

Reply:
Fixed, PTAL,thanks! @Jason918